server/controllers/automationStatsController.js

```diff
@@ -33,13 +33,25 @@ const queryCountResolver = queryCount => Number(queryCount[0].count);
  *
  * @returns {object} response reporting automation stats
  */
+
+const dateSubtractor = (id) => {
+  const date = id === 'days'
+    ? 0
+    : 1;
+  return date;
+};
+
 export default async (req, res) => {
-  const date = req.query.date
+  const { id } = req.params;
+  const dateEnd = req.query.date
     ? moment(req.query.date).format('YYYY-MM-DD')
     : moment().format('YYYY-MM-DD');
+  const dateStart = req.query.date
```

Comment:
Could you please explain what is happening here

Reply:
In order to get a duration for statistics, I need to calculate the timeframe for statistics, get the start and end date.
`dateEnd` is the current date or the date specified in the URL query.
`dateStart` is the period from which we start getting the statistics. So if a date is not specified in the URL query, I get the current date and subtract the `duration` from it.
However, if we want to get statistics for the current day i.e. `duration` = `days`. I specify the duration to subtract as 0 (this is what is being done in the `dateSubtractor` method).